### ISSUE TITLE
Simpler implementation

### ISSUE THREAD

**Issue by toomuchdesign** [STRML/babel-plugin-es2015-maybe-webpack#3]
Hi @STRML,
thanks for this plugin.

It's currently helping me to babelify unit tests tests with es6 modules enabled along with webpack 2 which currently needs es6 module transpilation turned off.

Unfortunately I was still not able make everything work properly, and I tried to have a look into your maybe-webpack plugin.

I found that It parses the exported babel config to remove -when needed- the specific modules preset. A lot of work!

Since babel [supports modules and loose options](https://babeljs.io/docs/plugins/preset-es2015/#options), I was able to dynamically disable es6 modules transpilation with:

```js
'use strict';
var babelPresetEs2015 = require('babel-preset-es2015');
var IS_WEBPACK2 = /^2\./.test(process.env.WEBPACK_VERSION);

module.exports = IS_WEBPACK2 === false
  ? babelPresetEs2015
  : babelPresetEs2015.buildPreset(null, {modules: false});
```
The trick here is that `babel-preset-es2015` returns a `buildPreset `method which is able to regenerate a preset configuration accepting the options specified in the link above. No parsing, no external dependencies.

Now babel is perfectly doing its job, and Webpack Hot module reloading is running great!

The downside here, is that we're relying on `babel-preset-es2015` interface, which may change in future.

Happy to here any feedback. Cheers!

**Comment by STRML** [comment]
Nice, that is very slick! I didn't realize `es2015` had exported a `modules` option, that seems to be new.

I'll update the module.

**Comment by toomuchdesign** [comment]
Me neither! It was added from 6.14.0 version. I'm not sure it's a documented feature.

I'll make a basic PR with updated implementation and peerDependencies just for reference.
Cheers!

**Comment by STRML** [comment]
Ah - I was so excited about shrinking the module that I already did it - released as `v1.2.0`. Thanks for the pointer!
